addon/components/select-2.js

```diff
@@ -69,6 +69,11 @@ var Select2Component = Ember.Component.extend({
     options.multiple = this.get('multiple');
     options.allowClear = this.get('allowClear');
     options.minimumResultsForSearch = this.get('searchEnabled') ? 0 : -1 ;
+    
+    // override select2's default id fetching behavior
+    options.id = (function(e) {
+      return e == undefined ? null : e.get('id');
```

Comment:
JSHint is looking for `e === undefined`. You should also probably use `e.id || e.get("id")` just to account for cases where non-models are being passed into Select2, so it defaults back to the built-in `id` property.